### ISSUE TITLE
[Feature]: Remove gogo from jaeger query service

### ISSUE THREAD

**Issue by gregorychen3** [jaegertracing/jaeger#4121]
### Requirement

External users may want to call jaeger query service to retrieve their traces to do a variety of things. Things like custom trace monitoring UIs, or for auditing purposes. This functionality is advertised in the Jaeger docs [here](https://www.jaegertracing.io/docs/1.23/apis/#trace-retrieval-apis).

### Problem

The jaeger query grpc api still depends on gogo proto, which is now deprecated (the last release was Jan 2021). This inconveniences the api user, as they must now use gogo as well to call the API.

Note that in most cases, gogo is cross compatible with vanially protobuf. However, this is not the case with jaeger query, as jaeger uses gogo's custom types for its [traceIDs](https://github.com/jaegertracing/jaeger-idl/blob/main/proto/api_v2/query.proto#L40). These gogo [docs](https://github.com/gogo/protobuf/blob/master/custom_types.md#warnings-and-issues) explain that gogo custom types are not interopable with vanilla proto in several ways.

So ultimately, the user has to use this long-deprecated gogo software, if they want to call jaeger query's grpc api.

### Proposal

Remove gogo from jaeger.

### Open questions

If maintainers agree that this work is needed, I would be happy to contribute. I have experience with gogo, grpc, and protobuf.

**Comment by yurishkuro** [comment]
I don't agree with the statement that internal use of gogo affects the users. If you're referring to the use of gogo annotations in the .proto files, then all that's needed is to remove those annotations and use .proto to generate plain gRPC code, without any dependency on gogo. Gogo annotations only affect the auto-generated code used internally in Jaeger, but they do not affect binary proto format.

Having said that, of course it would be great if we could remove the dependency on gogo. The problem is that I do not know of any equivalent that can provide the same performance characteristics, like mapping `[]byte` fields to two int64s (which we do for TraceID), or avoiding pointers when dealing with arrays.
